public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -332,9 +332,11 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
       }
 
       let queryObj;
-      if (target.isLogsQuery) {
+      if (target.isLogsQuery || queryDef.hasMetricOfType(target, 'logs')) {
```

Comment:
wonder if the explore logs query field can use the same query metric instead of isLogsQuery to simplify this logic.